### PR TITLE
API-4383: Added items needed for publishing get notifications endpoin…

### PR DESCRIPTION
**Pull request by mattclark-zerogravit**
…t and added auth connector to get client id from auth

### CODE REVIEW

File: resources/public/api/definition.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{
+  "scopes": [
```
Comment (lburgos) [inline resources/public/api/definition.json:2]: I don't think we need scopes as all the endpoints seem to be application restricted.

File: resources/public/api/definition.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    {
+      "key": "hello",
+      "name": "Say Hello",
+      "description": "Say Hello!"
+    }
+  ],
+  "api": {
+    "name": "Hello World",
```
Comment (lburgos) [inline resources/public/api/definition.json:10]: We should update these fields with more appropriate names.

File: app/uk/gov/hmrc/pushpullnotificationsapi/controllers/actionbuilders/AuthAction.scala [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+  override def executionContext: ExecutionContext = ec
+  override def refine[A](request: Request[A]):
+  Future[Either[Result, AuthenticatedNotificationRequest[A]]]  = {
+    implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromHeadersAndSession(request.headers)
+
+    authorised().retrieve(uk.gov.hmrc.auth.core.retrieve.v2.Retrievals.clientId) {
+      case maybeClientId: Option[String]  => maybeClientId match {
+        case Some(clientId) =>  Future.successful(Right(AuthenticatedNotificationRequest[A](clientId, request)))
+        case _ => Future.successful(Left(Unauthorized("unable to retrieve client id")))
```
Comment (lburgos) [inline app/uk/gov/hmrc/pushpullnotificationsapi/controllers/actionbuilders/AuthAction.scala:44]: Should we return a json object for these unauthorized responses as we do below on line 48?

File: resources/public/api/definition.json (hunk 1 above)
Comment (matevans) [reply resources/public/api/definition.json:2]: @lburgos I think as we are API Platform we should be setting a good example and also proving that scopes work with Bruce?

Comment (lburgos) [reply resources/public/api/definition.json:2]: Good point, let's see if we can get some value out of them and we can always remove them afterwards if they're pointless.

File: app/uk/gov/hmrc/pushpullnotificationsapi/controllers/actionbuilders/AuthAction.scala (hunk 3 above)
Comment (mattclark-zerogravit) [reply app/uk/gov/hmrc/pushpullnotificationsapi/controllers/actionbuilders/AuthAction.scala:44]: Will implement this in API-4394

**Pull request merged**